BranchMetrics/AdobeBranchExtension-iOS
# [SDK-1966][major] Migrated to AEPCore

## Reference
SDK-1966 -- Update AdobeLaunchExtension to support MobileCore versions 2+ 

## Summary
This PR reflects updates to the Adobe SDK, which has transitioned from the Adobe Cloud Platform (ACP) to the Adobe Experience Platform (AEP). This transition from ACP to AEP involves a major update, including a shift from Objective-C to Swift, resulting in changes to class names and methods. This PR also includes updating the BranchSDK from 1.43.2 to 2.1.2.


**[Link to Adobe's documentation on the changes can be found here ](https://developer.adobe.com/client-sdks/previous-versions/documentation/migrate-to-swift/)**

#### Key changes include:
- Transition from `ACPExtensionEvent` to `AEPEvent`
- Transition from `ACPExtension` to `AEPExtension` and new protocol methods
- Changes in the way listeners are created
- Changes in the way extensions are registered and initialized

## Motivation
Clients who want to take advantage of the new AEP SDK's requested that we update our extension to be compatible. 

## Type Of Change


## Testing Instructions
To verify these changes:

1. Check out the repository and integrate it into a sample app or use the existing AdobeBranchExample app.
2. Ensure that all events and Adobe IDs are correctly passed through to Branch.
3. Verify that all of the previous functionalities of the extension remain intact.










cc @BranchMetrics/saas-sdk-devs for visibility.
